test/unary_operator_test.js

```diff
@@ -40,17 +40,38 @@ describe('unary operators', () => {
     check(`typeof a`, `typeof a;`);
   });
 
-  it.skip('converts unary existential identifier checks to typeof + null check', () => {
+  it('converts unary existential identifier checks to typeof + null check', () => {
     check(`a?`, `typeof a !== 'undefined' && a !== null;`);
   });
 
-  it.skip('converts unary existential non-identifier to non-strict null check', () => {
+  it('converts unary existential non-identifier to non-strict null check', () => {
     check(`a.b?`, `a.b != null;`);
     check(`0?`, `0 != null;`);
   });
 
-  it.skip('surrounds unary existential operator results if needed', () => {
+  it('surrounds unary existential operator results if needed', () => {
     check(`a? or b?`, `(typeof a !== 'undefined' && a !== null) || (typeof b !== 'undefined' && b !== null);`);
     check(`0? or 1?`, `(0 != null) || (1 != null);`);
+    check(`(0?) or (1?)`, `(0 != null) || (1 != null);`);
+  });
+
+  it('converts unary existential operator and handles negation', () => {
+    check(`
+      unless a? and c
+        b
+    `, `
+      if ((typeof a === 'undefined' || a === null) || !c) {
+        b;
+      }
+    `);
+
+    check(`
+      unless a.b?
+        b
+    `, `
+      if ((a.b == null)) {
```

Comment:
Ah. That's probably beyond the scope of this PR. Let's just test for the
double parens. What is the question about compound expressions?
On Thu, Feb 25, 2016 at 8:25 PM Jeff Chan <notifications@github.com> wrote:

> In test/unary_operator_test.js
> <https://github.com/decaffeinate/decaffeinate/pull/127#discussion_r54203915>
> :
>
> > +
> > +  it('converts unary existential operator and handles negation', () => {
> > +    check(`
> > +      unless a? and c
> > +        b
> > +    `, `
> > +      if ((typeof a === 'undefined' || a === null) || !c) {
> > +        b;
> > +      }
> > +    `);
> > +
> > +    check(`
> > +      unless a.b?
> > +        b
> > +    `, `
> > +      if ((a.b == null)) {
>
> The double parens is the last piece that I'm struggling with. Do I need to
> detect that the conditional has no surrounding parens, then I can assume
> that ConditionalPatcher will take care of it, so I should not insert the
> parens? What about compound expressions?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/decaffeinate/decaffeinate/pull/127/files#r54203915>.
>